### PR TITLE
[15.0][MIG] fieldservice_route: Migration to 15.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 15.0

Please @pedrobaeza and @CarlosRoca13 can you review it?

@Tecnativa TT35715

### CODE REVIEW

**Review by dreispt — APPROVED** [review]

File: fieldservice_route/models/fsm_order.py [hunk 1]
```diff
@@ -14,11 +14,16 @@ class FSMOrder(models.Model):
...
+        comodel_name="fsm.route.dayroute", string="Day Route", index=True
+    )
     fsm_route_id = fields.Many2one(related="location_id.fsm_route_id", string="Route")
 
     person_id = fields.Many2one(
-        "fsm.person", string="Assigned To", index=True, default=_get_default_person
+        comodel_name="fsm.person",
+        string="Assigned To",
+        index=True,
+        default=_get_default_person,
```
Comment (pedrobaeza) [inline fieldservice_route/models/fsm_order.py]: Please use lambda

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please change all the default to lambdas

**Comment by dreispt** [comment]
@pedrobaeza why would lambda usage be required?

**Comment by pedrobaeza** [comment]
For having each time evaluated the value, instead of only on instance initialization. See also the guidelines for that: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#37models

File: fieldservice_route/models/fsm_order.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Copyright (C) 2019 Open Source Integrators
 # Copyright (C) 2019 Serpent consulting Services
+# Copyright 2022 Tecnativa - Víctor Martínez
```
Comment (nikul-serpentcs) [inline fieldservice_route/models/fsm_order.py]: @victoralmau Thank you for PR.

Can you please remove your **Copyright** and add in the contributor's fragment?

Comment (victoralmau) [reply fieldservice_route/models/fsm_order.py]: Changes done.

Comment (pedrobaeza) [reply fieldservice_route/models/fsm_order.py]: If he changes the file, the copyright should be there.

File: fieldservice_route/models/fsm_order.py [hunk 3]
```diff
@@ -14,13 +14,23 @@ class FSMOrder(models.Model):
     def _get_default_person(self):
         return self.fsm_route_id.fsm_person_id.id or False
```
Comment (pedrobaeza) [inline fieldservice_route/models/fsm_order.py]: This is not going to work, as `fsm_route_id` won't never have a value on default. This must work with context values or previous defaults.

File: fieldservice_route/models/fsm_order.py [hunk 4]
```diff
@@ -14,13 +14,23 @@ class FSMOrder(models.Model):
...
     person_id = fields.Many2one(
-        "fsm.person", string="Assigned To", index=True, default=_get_default_person
+        comodel_name="fsm.person",
+        string="Assigned To",
+        index=True,
+        default=lambda self: self._get_default_person(),
     )
 
+    @api.onchange("fsm_route_id")
+    def _onchange_person_id(self):
```
Comment (pedrobaeza) [inline fieldservice_route/models/fsm_order.py]: Can you do it a computed writable field instead?

**Review by pedrobaeza — COMMENTED** [review]
Please squash the initial commits from Maxime that are the same into one.

**Comment by victoralmau** [comment]
Changes done.

**Comment by pedrobaeza** [comment]
The converted computed field is not computed writable, and I still see a lot of repeated commits from Maxime.

**Comment by victoralmau** [comment]
> The converted computed field is not computed writable, and I still see a lot of repeated commits from Maxime.

Sorry but the `person_id` field is now compute `store=True`, right? (although in `fieldservice` addon it is many2one )

I have updated the commit history to reduce Maxime's commits.

**Comment by pedrobaeza** [comment]
But it misses readonly=False

**Comment by victoralmau** [comment]
> But it misses readonly=False

Ok, changes done.

**Review by CarlosRoca13 — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**